### PR TITLE
Incident response & Triage docs

### PR DESCRIPTION
**Pull request by MatthewJWalls**
## What does this change?

WIP, still on first draft.

The objective of writing this document is to:

* Be useful for helping people in our team respond to and investigate both minor and major Frontend incidents (site outages)
* Give advice on how to go about investigating issues on the Frontend stack
* Define where all of our sources of truth / dashboards are

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [ ] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: docs/01-start-here/08-incidents.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+All of our Frontend metrics, alerts and monitoring are available in two places: 
+
+* [CloudWatch](https://eu-west-1.console.aws.amazon.com/cloudwatch/) for health and performance metrics
+* [Kibana](https://logs.gutools.co.uk/app/kibana) for application logs
+
+You may also want to look at the [Fastly fashboard](https://manage.fastly.com/)
+
+## Process and roles
+
+Response to a potential site incidents will involve in general the following steps
```
Comment (NataliaLKB) [inline docs/01-start-here/08-incidents.md]: Hey thanks for writing this up. Can you include this doc which @nicl and I started as well? https://docs.google.com/document/d/1sAq378Oqm5NUG2_FJORDSd_Tag6gUUUsZaE9zUsgWHc/edit

Comment (MatthewJWalls) [reply docs/01-start-here/08-incidents.md]: Sweet that means I can pull out some of the process parts and it'll make things a bit more focused here. Doc is still basically notes at this point will be updating for a while.

Comment (NataliaLKB) [reply docs/01-start-here/08-incidents.md]: Thanks @MatthewJWalls ! It is a great start. Very useful thing to be in our docs

File: docs/01-start-here/08-incidents.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+techniques for finding the cause of issues with the Frontend stack.
+
+## Sources of Truth
+
+All of our Frontend metrics, alerts and monitoring are available in two places: 
+
+* [CloudWatch](https://eu-west-1.console.aws.amazon.com/cloudwatch/) for health and performance metrics
+* [Kibana](https://logs.gutools.co.uk/app/kibana) for application logs
+
+You may also want to look at the [Fastly fashboard](https://manage.fastly.com/)
```
Comment (NataliaLKB) [inline docs/01-start-here/08-incidents.md]: dashboard not fashboard

**Comment by MatthewJWalls** [comment]
@nicl @QuarpT @philmcmahon any feedback would be appreciated, I'm sure there's much more we can add

File: docs/01-start-here/08-incidents.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+CAPI, then it will be reported as 'CAPI being slow' even though the latency is within our own 
+app.
+
+### Analysing Instances Directly
+
+If you suspect that there is a real issue with an app you can connect directly to the instance
+using to try and collect more runtime information. 
+
+You should start by picking an instance and detaching it from the load balancer so that it doesn't
+get killed or interfered with, and then connect to via ssm.
```
Comment (philmcmahon) [inline docs/01-start-here/08-incidents.md]: Maybe it's worth running top/lsof before detaching from the ELB - to see what the app looks like under load

**Comment by philmcmahon** [comment]
Nice!

File: docs/01-start-here/08-incidents.md (hunk 3 above)
Comment (MatthewJWalls) [reply docs/01-start-here/08-incidents.md]: good point

File: docs/01-start-here/08-incidents.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+You should start by picking an instance and detaching it from the load balancer so that it doesn't
+get killed or interfered with, and then connect to via ssm. 
+
+You should look for things like:
+
+* The application logs
+* Running ```top``` to get memory and cpu usage
+* Check diskspace with ```df -h```
+* Get the number of open file descriptors with ```lsof | wc -l```
```
Comment (QuarpT) [inline docs/01-start-here/08-incidents.md]: You could also include running jstack to get a thread dump from a jvm process

File: docs/01-start-here/08-incidents.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+## Cleaning up after the incident is over
+
+[Keep following the P1 checklist for communications and process information after the incident is resolved](https://docs.google.com/document/d/1sAq378Oqm5NUG2_FJORDSd_Tag6gUUUsZaE9zUsgWHc/edit?usp=sharing)
+
+If you scaled frontend apps up, you should wait for latency and error rates have returned to normal
+before scaling back down again.
```
[the remark this answers is not in the document]
Comment (NataliaLKB) [reply docs/01-start-here/08-incidents.md]: It's not usually worth scaling down manually (unless you scaled up to over 18 instances). I would normally just recommend people let auto scaling down do its work

**Comment by NataliaLKB** [comment]
Is this still WIP ?

**Comment by MatthewJWalls** [comment]
I'm going to merge this. I think doing it via the github editor has meant the commit triggers to regenerate the ToC hasn't run properly so I'll redo it locally and force push this.

**Pull request merged**